pkg/apis/pipeline/v1alpha1/pull_request_resource.go

```diff
@@ -0,0 +1,159 @@
+/*
+Copyright 2018 The Knative Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+import (
+	"flag"
+	"fmt"
+	"strings"
+
+	"github.com/tektoncd/pipeline/pkg/names"
+	corev1 "k8s.io/api/core/v1"
+)
+
+var (
+	prSource = "pr-source"
+	// The container that we use to implement the PR source step.
+	prImage = flag.String("pr-image", "override-with-pr:latest",
+		"The container image containing our PR binary.")
+)
+
+// PullRequestResource is an endpoint from which to get data which is required
+// by a Build/Task for context.
+type PullRequestResource struct {
+	Name             string               `json:"name"`
+	Type             PipelineResourceType `json:"type"`
+	URL              string               `json:"url"`
+	GithubOauthToken string               `json:"githubOauthToken"`
```

Comment:
is the plan to support more than just oauth token based auth? (disclaimer: i dont know what im talking about :sweat_smile: )